core/controllers/acl_decorators.py

```diff
@@ -3924,7 +3924,7 @@ def test_can_accept_suggestion(
             target_id: str,
             suggestion_id: str,
             **kwargs: Any
-        ) -> _GenericHandlerFunctionReturnType:
+        ) -> None:
```

Comment:
Reading the implementation, this clearly returns something, so how can the return type be None?

Reply:
This `get_decorator_for_accepting_suggestion` decorator is only used for HTTP methods and it is used in only 3 places, and we know HTTP methods always return none:
 
https://github.com/oppia/oppia/blob/5471941da48071723b5f7e8684716584c08c6a1d/core/controllers/suggestion.py#L146
https://github.com/oppia/oppia/blob/5471941da48071723b5f7e8684716584c08c6a1d/core/controllers/suggestion.py#L216
https://github.com/oppia/oppia/blob/5471941da48071723b5f7e8684716584c08c6a1d/core/controllers/acl_decorators_test.py#L3535

> so how can the return type be None?

this `test_can_accept_suggestion` method return the value of decorated function not the function itself and here the decorated function is handler. So I think docstring need to be updated as well!.

Reply:
> Why is this treated in a different way to the other similar functions in this file?

Because previously `get_decorator_for_accepting_suggestion` decorator is defined like this:
```python
def get_decorator_for_accepting_suggestion(
    decorator: Callable[..., Callable[..., _GenericHandlerFunctionReturnType]]
) -> Callable[..., Callable[..., _GenericHandlerFunctionReturnType]]:...
```
and from implementation, we can see that this decorator is accepting another decorator as an argument, and in acl_decorator.py file, every decorator has a Generic return type. for eg: `acl_decorators.can_edit_exploration`

```python
def can_edit_exploration(
    handler: Callable[..., _GenericHandlerFunctionReturnType]
) -> Callable[..., _GenericHandlerFunctionReturnType]: ...
```

So, when we pass this `can_edit_exploration` decorator to `get_decorator_for_accepting_suggestion` decorator then __GenericHandlerFunctionReturnType is figured as again __GenericHandlerFunctionReturnType by MyPy because in this case `can_edit_exploration` has `Callable[..., Callable[..., _GenericHandlerFunctionReturnType]]` type and decorator argument also has  Callable[..., Callable[..., _GenericHandlerFunctionReturnType]] type. So, due to this actual return type is never figured out.

But in the case of other decorators, we are providing Http methods directly to the decorators. and every Http method has a None return type so __GenericHandlerFunctionReturnType is figured as None.

This is the only decorator in the whole file which accepting another decorator as an argument.

Reply:
Ah ok, thanks! Sounds good, I had missed that this wasn't itself a decorator.

Reply:
Ah ok, thanks! Sounds good, I had missed that this wasn't itself a decorator.